index.bs

```diff
@@ -9589,6 +9619,76 @@ The [=remote end steps=] given |session|, and |command parameters| are:
 
 </div>
 
+### Events ### {#module-input-events}
+
+#### The input.fileDialogOpened Event #### {#event-input-fileDialogOpened}
+
+<dl>
+   <dt>Event Type</dt>
+   <dd>
+      <pre class="cddl local-cddl remote-cddl">
+         input.FileDialogOpened = (
+            method: "input.fileDialogOpened",
+            params: input.FileDialogInfo
+         )
+
+         input.FileDialogInfo = {
+            context: browsingContext.BrowsingContext,
+            element: script.SharedReference,
+            multiple: bool,
+         }
+      </pre>
+   </dd>
+</dl>
+
+<div algorithm>
+
+The [=remote end event trigger=] is the <dfn export>WebDriver BiDi file dialog
+opened</dfn> steps given |element|.
+
+1. Let |context| be the |element|'s [=node document=]'s [=/browsing context=].
+
+1. Let |context id| be the [=browsing context id=] for |context|.
+
+1. Let |related browsing contexts| be a [=/set=] containing |context|.
+
+1. Let |sessions| be the [=set of sessions for which an event is enabled=]
+   given <code>"input.fileDialogOpened"</code> and |related browsing
+   contexts|.
+
+1. Let |params| be a [=/map=] matching the <code>input.FileDialogInfo</code>
+   production, with the <code>context</code> field set to |context id|, the
+   <code>element</code> field set to |element|, and the <code>multiple</code>
+   field set to true if the [=multiple=] attribute is set on |element| and false
+   otherwise.
+
+1. Let |body| be a [=/map=] matching the <code>input.FileDialogOpened</code>
+   production, with the <code>params</code> field set to |params|.
+
+1. For each |session| in |sessions|:
+
+  1. [=Emit an event=] with |session| and |body|.
+
+1. Let |intercepting sessions| be the [=set of sessions for which file dialog interception is enabled=].
```

Comment:
I don't think this should use "did any session suppress the dialog". Rather if any session actually requests a specific capability value here then any later session requesting a different capability value should fail during session creation. If sessions don't request a specific value then a later session requesting a specific value can affect the behaviour (maybe later we should have a `session.CapabilityChanged` event for when a new session affects existing settings, but that's out of scope for now).